packages/vite/src/node/plugins/manifest.ts

```diff
@@ -114,18 +114,36 @@ export function manifestPlugin(config: ResolvedConfig): Plugin {
 
       const cssEntryFiles = cssEntryFilesCache.get(config)!
 
+      const fileNameToAssetMeta = new Map<string, { originalName: string }>()
+      generatedAssets.get(config)!.forEach((asset, referenceId) => {
+        const fileName = this.getFileName(referenceId)
+        fileNameToAssetMeta.set(fileName, asset)
+      })
```

Comment:
Maybe moving this above `function createAsset` and including `const src = fileNameToAssetMeta.get(chunk.fileName)?.originalName ?? chunk.name` in `createAsset` would remove `fileNameToAsset`?

Reply:
I tried this, but it means we need to return `{ asset, src }` from `createAsset` except if we also move the assignment to the manifest. And IIUC, we still need `fileNameToAsset` to include the duplicates. Would you propose a changeset? Maybe I'm missing something here.

Reply:
You're right. My bad. I was mixing thing up in my head.